### PR TITLE
Feature/stateless validation

### PR DESCRIPTION
**Pull request by kamilsa**
Stateless validation that checks:

1. If signatures are correct (uses crypto-provider)
2. If tx is not outdated (checks if tx was created in)
3. If tx is from future

Also stubs for crypto-provider and stateless validation were removed and replaced by implementation

### CODE REVIEW

File: libs/model/model_crypto_provider_impl.cpp [hunk 1]
```diff
@@ -15,43 +15,44 @@
...
 #include <model/model_crypto_provider_impl.hpp>
 #include <model/model_hash_provider_impl.hpp>
 
 namespace iroha {
   namespace model {
 
-  ModelCryptoProviderImpl::ModelCryptoProviderImpl(ed25519::privkey_t privkey,
-  ed25519::pubkey_t pubkey)
-  : privkey_(privkey), pubkey_(pubkey) {}
+    ModelCryptoProviderImpl::ModelCryptoProviderImpl(ed25519::privkey_t privkey,
```
Comment (motxx) [inline libs/model/model_crypto_provider_impl.cpp]: So now, should tx be validated when stateless validation?

File: libs/model/model_crypto_provider_impl.cpp [hunk 2]
```diff
@@ -15,43 +15,44 @@
...
+      for (auto sign : tx.signatures) {
+        auto verified = iroha::verify(tx_hash.data(), tx_hash.size(),
+                                      sign.pubkey, sign.signature);
+        if (!verified) return false;
+      }
+      return true;
+    }
 
-    auto sign = iroha::sign(tx_hash.data(), tx_hash.size(), pubkey_, privkey_);
+    Transaction &ModelCryptoProviderImpl::sign(Transaction &tx) {
```
Comment (motxx) [inline libs/model/model_crypto_provider_impl.cpp]: Is this used by client, not in Iroha?

File: irohad/validation/stateless/validator_impl.hpp [hunk 3]
```diff
@@ -15,21 +15,24 @@
...
+    class StatelessValidatorImpl : public StatelessValidator {
      public:
-      StatelessValidatorStub();
-      bool validate(const model::Transaction &transaction) const override;
-    };
+      explicit StatelessValidatorImpl(model::ModelCryptoProvider& crypto_provider);
+      bool validate(const model::Transaction& transaction) const override;
 
+     private:
+      uint64_t max_delay_;  // max-delay between tx creation and validation
```
Comment (motxx) [inline irohad/validation/stateless/validator_impl.hpp]: Can it be const or constexpr?

Comment (kamilsa) [inline irohad/validation/stateless/validator_impl.hpp]: yes, sure

File: libs/model/model_crypto_provider_impl.cpp (hunk 2 above)
Comment (kamilsa) [inline libs/model/model_crypto_provider_impl.cpp]: You are right. Client should sign transactions, not Iroha. So this method is not needed.

**Comment by motxx** [comment]
I don't know whether validator should be func or class, but other changes that I've not commented look good for me.

**Pull request merged**